jobs: TestJitterCalculation failed

jobs.TestJitterCalculation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16347664?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16347664?buildTab=artifacts#/) on master @ [9915bd965cab50ff5e7c9214447a360deec67cd5](https://github.com/cockroachdb/cockroach/commits/9915bd965cab50ff5e7c9214447a360deec67cd5):


```
=== RUN   TestJitterCalculation
--- FAIL: TestJitterCalculation (0.00s)
=== RUN   TestJitterCalculation/small
    registry_test.go:1236: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:1236
        	Error:      	Should not be: 1000000
        	Test:       	TestJitterCalculation/small
    --- FAIL: TestJitterCalculation/small (0.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestJitterCalculation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-40985

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -710,13 +710,21 @@ func TestJitterCalculation(t *testing.T) {
 		},
 	} {
 		t.Run(test.name, func(t *testing.T) {
-			interval := jitter(test.input)
+			interval := test.input
+			testutils.SucceedsSoon(t, func() error {
+				interval = jitter(test.input)
+				// This check will sometimes fail due to a randomly picking a value
+				// that rounds to 1/2. The frequency of this is a bit higher than one
+				// would expect because the rounding behavior depends on the width of
+				// the duration, not the width of a float64.
+				if interval == test.input && test.input != 0 {
+					return errors.New("jitter returned same value as input")
+				}
+				return nil
+			})
 			rangeMin, rangeMax := outputRange(test.input)
 			require.GreaterOrEqual(t, rangeMax, interval)
 			require.LessOrEqual(t, rangeMin, interval)
-			if test.input != 0 {
-				require.NotEqual(t, test.input, interval)
-			}
 		})
 	}
 }

